source/base/polynomials_piecewise.cc

```diff
@@ -139,7 +139,16 @@ namespace Polynomials
         else
           {
             const double offset = step * interval;
-            if (x < offset || x > offset + step)
+            // ROCm 5.7 throws a floating point exception in debug mode when
+            // trying to evaluate (x < offset && x > offset + step). Separating
```

Comment:
You have a mistake in the comment where you say `&&`. You meant to say `||`, and with the current state of the comment the separation into the two `if` parts does not make sense...